packages/lesswrong/server/callbacks/compileEbook.ts

```diff
@@ -0,0 +1,308 @@
+//         console.log(err)
+//       })  
+//   } else if (config.sequenceId) {
+//     const query = `{
+//       sequence(input: {selector: {_id: "${config.sequenceId}"}}) {
+//         result {
+//           title
+//             contents {
+//               html
+//             }
+//             chapters {
+//               title
+//               contents {
+//                 html
+//               }
+//               posts {
+//                 title
+//                 contents {
+//                   html
+//                 }
+//                 _id
+//               }
+//             }
+//         }
+//       }
+//     }
+//     `
+//     request('https://www.lessestwrong.com/graphql?', query)
+//       .then(data => {
+//         buildEbookFromSequence(data.sequence.result)
+//         epub.writeEPUB(console.error, 'output', config.shorttitle, ()=>{
+//           console.log('Book successfully written to output/' + config.shorttitle + '.epub');
+//         });
+//       })
+//       .catch(err => {
+//         console.log(err)
+//       })  
+//   }
+// }
+
+const getConfigFromSequence = (sequence: DbSequence, author: DbUser, imagePath: string) => {
+  return {
+      id: slugify(sequence.title),
+      title: sequence.title,
+      cover: imagePath,
+      series: siteNameWithArticleSetting.get(),
+      sequence: 1,
+      author: author.displayName,
+      fileAs: author.displayName,
+      // TODO come back to
+      // genre: "Non-Fiction",
+      // tags: "Rationality",
+      // copyright: `${author.displayName}, `, "Scott Alexander, 2017",
+      publisher: siteNameWithArticleSetting.get(),
+      // published: "2017",
+      language: "en",
+      description: "",
+      contents: "Chapters",
+      source: getSiteUrl(),
+      images: []
+  }
+}
+
+const htmlTemplate = `
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
+  <head>
+    <meta http-equiv="Content-Type" content="application/xhtml+xml; charset=utf-8" />
+    <link rel="stylesheet" href="style/base.css" />
+  </head>
+<body></body>
+</html>
+`
+
+export async function ChaptersEditEbookCallback (chapter: DbChapter) {
+  console.log("BEGIN ChaptersEditEbookCallback")
+  const context = await createAdminContext();
+  const posts = await sequenceGetAllPosts(chapter.sequenceId, context)
+  const chapters = await Chapters.find({sequenceId: chapter.sequenceId}).fetch()
+  const sequence = await Sequences.findOne({_id: chapter.sequenceId})
+  const author = await Users.findOne({_id: sequence?.userId})
+
+  if (!sequence) throw Error("No sequence found")
+  if (!author) throw Error("No author found")
+
+  // get cover image
+  const imageUrl = makeCloudinaryImageUrl(sequence.bannerImageId, {f:'jpg'})
+  const file = fs.createWriteStream("tmp/ebookImage0.jpg")
+  
+  https.get(imageUrl, (response) => {
```

Comment:
You probably want this to be `await`-style rather than callback-style, making an adapter utility function if necessary.